Sources/Xcodeproj/print().swift

```diff
@@ -0,0 +1,186 @@
+        try write("        };")
+        try write("        \(productBuildConfiguration(module)) = {")
+        try write("            isa = XCBuildConfiguration;")
+        try write("            buildSettings = { \(module.buildSettings) };")
+        try write("            name = Debug;")
+        try write("        };")
+        //TODO ^^ probably can consolidate this into the three kinds
+        //TODO we use rather than have one per module
+
+        // targets that depend on this target use these
+        try write("        \(targetDependency(module)) = {")
+        try write("            isa = PBXTargetDependency;")
+        try write("            target = \(productsGroupRef(module));")
+        // try write("            targetProxy = \(targetDependencyProxy(module));")
+        try write("        };")
+        // try write("        \(targetDependencyProxy(module)) = {")
+        // try write("            isa = PBXContainerItemProxy;")
+        // try write("            containerPortal = REF00000000;")
+        // try write("            proxyType = 1;")
+        // try write("            remoteGlobalIDString = \(productsGroupRef(module));")
+        // try write("            remoteInfo = \(module.c99name);")
+        // try write("        };")
+    }
+
+////// “Sources” group
+    try write("        REF0000000b = {")
+    try write("            isa = PBXGroup;")
+    try write("            children = (" + nontests.map(moduleGroupName).joinWithSeparator(", ") + ");")
+    try write("            name = Sources;")
+    try write("            sourceTree = '<group>';")
+    try write("        };")
+
+////// “Tests” group
+    try write("        REF0000000c = {")
+    try write("            isa = PBXGroup;")
+    try write("            children = (" + tests.map(moduleGroupName).joinWithSeparator(", ") + ");")
+    try write("            name = Tests;")
+    try write("            sourceTree = '<group>';")
+    try write("        };")
+
+////// products group
+    try write("        REF00000017 = {")
+    try write("            isa = PBXGroup;")
+    try write("            children = (" + modules.map(productRef).joinWithSeparator(", ") + ");")
+    try write("            name = Products;")
+    try write("            sourceTree = '<group>';")
+    try write("        };")
+
+////// build configuration
+    try write("        REF00000003 = {")
+    try write("            isa = XCBuildConfiguration;")
+    try write("            buildSettings = {};")
+    try write("            name = Debug;")
+    try write("        };")
+    try write("        REF00000008 = {")
+    try write("            isa = XCConfigurationList;")
+    try write("            buildConfigurations = (REF00000003);")
+    try write("            defaultConfigurationIsVisible = 0;")
+    try write("            defaultConfigurationName = Debug;")
+    try write("        };")
+    try write("    };")
+    try write("    rootObject = REF00000000;")
+    try write("}")
+}
```

Comment:
Ah, sorry for not being clear (I think I was replying to a comment of yours?). Specifically, I thought it would be nice to explain what `buildActionMask` is, and why it takes the value `2147483647`.

Also, after taking another look at this, I wonder if it would be nicer to store object references like `REF00000003` in variables? That would make it easier to track and modify which objects are included in which groups, etc. Just a thought!